src/rsa.rs

```diff
@@ -475,6 +486,51 @@ extern {
                                  rng: *mut rand::RAND) -> c::int;
 }
 
+#[cfg(feature = "rsa_signing")]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn BN_rand_range(dest: *mut BIGNUM, range: *const BIGNUM,
+                                   rng: *mut rand::RAND) -> c::int {
```

Comment:
Hmm. It seems like what you're suggesting a few differences from this PR, which could be taken separately:

(A) Instead of converting bytes to limbs each time we generate a number, we fill the limbs buffer directly. I like that: seems more efficient, requires one constant-time function fewer, and lets us avoid a few fishily-sized stack buffers. However, it's not immediately obvious to me that even so we'll be able to avoid writing new code to handle the case of range bounds which use only a few bits of the most significant limb, which motivated most of the code that I added in `limb.rs`. FWIW I suspect a lot of that code for counting leading zero bytes/bits etc could be shortened by somebody more familiar with bitwise APIs and tricks.

(B) We leave the definition of BN_rand_range in C, and call into a Rust function internally. I'm neutral on this; it seems the advantage is we have a smaller diff, and the disadvantage is we can't delete `random.c` yet. If it helps us make (A) work, though, I'm for it.

(C) We start by writing a new key-generation function, then merge them later, instead of refactoring first. My gut reaction is against that. What's the advantage?